src/yt-music/YTProgressUpdates.ts

```diff
@@ -0,0 +1,73 @@
+import { Page } from 'puppeteer';
+import { Progress, ProgressUpdate, ProgressUpdateSubscriber } from '../progress-bar/ProgressUpdate';
+
+// This is pretty much the same as YTPlayUpdates
+
+export class YTProgressUpdates implements ProgressUpdate {
+  private subscribers: ProgressUpdateSubscriber[];
+
+  private progress: Progress;
+
+  public get currentProgress() {
+    return this.progress;
+  }
+
+  constructor(private page: Page, initialSubscribers?: ProgressUpdateSubscriber[]) {
+    this.subscribers = initialSubscribers || [];
+    this.handleProgressUpdate();
+  }
+  
+
+  public subscribe = (callback: ProgressUpdateSubscriber) => {
+    this.subscribers.push(callback);
+  };
+
+  public async forceProgressUpdate() {
+    
+    await this.page.waitForSelector(`video`);
+
+    const progress = await this.page.$eval(
+        `video`,
+        (element: HTMLVideoElement) => {
+          return { 
+            currentTime: element.currentTime, 
+            currentDuration: element.duration, 
```

Comment:
Does the song that is playing render as a video in the HTML? If this is just using the video that plays when a song is first selected, remember that in the future we want the user to be able to conduct searches and stuff while the music still plays (from the sound bar at the bottom of the page).

Reply:
Yes it does. There should only be one video element at a time. 
It should still work when we add user searches and we get the song list from the search because of the mini-player feature using the same video element.